### PR TITLE
Purge notifications to email reply to rows older than 7 days

### PR DESCRIPTION
**Pull request by kentsanggds**
When we purge the notifications table we also need to purge any rows in the link table `notifications_email_reply_to` table as we will not be able to remove the notifications with the foreign key constraint.

Reference 

Step 5 - https://www.pivotaltracker.com/story/show/150440099

### CODE REVIEW

File: app/celery/scheduled_tasks.py [hunk 1]
```diff
@@ -119,6 +120,17 @@ def delete_sms_notifications_older_than_seven_days():
...
+        current_app.logger.info(
+            "Delete {} job started {} finished {} deleted {} notification to email reply to mappings".format(
+                'email',
+                start,
+                datetime.utcnow(),
+                deleted_noti_to_reply_to
+            )
+        )
+
+        start = datetime.utcnow()
```
Comment (richardc0) [inline app/celery/scheduled_tasks.py]: Is this needed again there is already a similar line on 122

File: app/dao/notifications_dao.py [hunk 2]
```diff
@@ -26,14 +26,15 @@
...
     NotificationHistory,
     NotificationStatistics,
-    NotificationEmailReplyTo,
-    ServiceEmailReplyTo,
     ScheduledNotification,
+    ServiceEmailReplyTo,
     Template,
     KEY_TYPE_NORMAL,
     KEY_TYPE_TEST,
+    EMAIL_TYPE,
```
Comment (richardc0) [inline app/dao/notifications_dao.py]: Could be moved to alphabetical order.

File: app/dao/notifications_dao.py [hunk 3]
```diff
@@ -377,6 +378,22 @@ def delete_notifications_created_more_than_a_week_ago_by_type(notification_type)
...
+    subq = db.session.query(Notification.id).filter(
+        func.date(Notification.created_at) < seven_days_ago
+    ).subquery()
+    deleted = db.session.query(
+        NotificationEmailReplyTo
+    ).filter(
+        NotificationEmailReplyTo.notification_id.in_(subq)
+    ).delete(synchronize_session='fetch')
+    db.session.commit()
+    return deleted
```
Comment (richardc0) [inline app/dao/notifications_dao.py]: Might be worth checking the subq length = the deleted length and logging if there is a difference.

or do a synchronize_session='evaluate' check the number match then run the actual delete?

File: app/dao/notifications_dao.py (hunk 2 above)
Comment (kentsanggds) [reply app/dao/notifications_dao.py]: ok

File: app/dao/notifications_dao.py (hunk 3 above)
Comment (kentsanggds) [reply app/dao/notifications_dao.py]: Will add this to `delete_notifications_created_more_than_a_week_ago_by_type` function to limit impact when there is an issue when deleting records

**Pull request merged**